inspirehep/modules/records/receivers.py

```diff
@@ -75,122 +87,163 @@ def normalize_field_categories(sender, *args, **kwargs):
+
+      - Otherwise:
+        - If the record has no `start_page` and `artid` key in any of
+          its `publication_info`s then add 'preprint' to the document types.
+      - If the record has as a primary collection a key in the following
+        table then add the corresponding value to the document types.
+        All values found are added.
+
+          + -------- + -------- +
+          | key      | value    |
+          + -------- + -------- +
+          | lectures | lectures |
+          | review   | review   |
+          + -------- + -------- +
+
     """
-    experiments = json.get("accelerator_experiments")
-    if experiments:
-        for exp in experiments:
-            # FIXME: These lists are temporary. We should have a list of experiment names
-            # that is generated from the current state of our data.
-            from .experiment_list import EXPERIMENTS_NAMES as experiments_list_original, experiments_list
-            facet_experiments_list = []
-            experiments = exp.get("experiment")
-            if experiments:
-                if type(experiments) is not list:
-                    experiments = [experiments]
+    def _was_not_published(json):
+        def _not_published(publication_info):
+            return 'page_start' not in publication_info and 'artid' not in publication_info
+
+        publication_infos = force_force_list(get_value(json, 'publication_info'))
+        not_published = map(_not_published, publication_infos)
+
+        return all(not_published)
+
+    EXCLUSIVE_DOC_TYPE_MAP = {
+        'published': 'peer reviewed',
+        'thesis': 'thesis',
+        'book': 'book',
+        'bookchapter': 'book chapter',
+        'proceedings': 'proceedings',
+        'conferencepaper': 'conference paper',
+        'note': 'note',
+        'report': 'report',
+        'activityreport': 'activity report',
+    }
+
+    INCLUSIVE_DOC_TYPE_MAP = {
+        'lectures': 'lectures',
+        'review': 'review',
+    }
+
+    if 'collections' not in json:
+        json['facet_inspire_doc_type'] = []
+        return
+
+    result = []
+
+    primary_collections = force_force_list(get_value(json, 'collections.primary'))
+    normalized_collections = map(lambda el: el.lower(), primary_collections)
+
+    for collection in normalized_collections:
+        if collection in EXCLUSIVE_DOC_TYPE_MAP:
+            result.append(EXCLUSIVE_DOC_TYPE_MAP[collection])
+            break
+
+    if not result:
+        if _was_not_published(json):
+            result.append('preprint')
+
+    for collection in normalized_collections:
+        if collection in INCLUSIVE_DOC_TYPE_MAP:
+            result.append(INCLUSIVE_DOC_TYPE_MAP[collection])
+
+    json['facet_inspire_doc_type'] = result
+
+
+def match_valid_experiments(sender, json, *args, **kwargs):
+    """TODO."""
+    def _normalize(experiment):
+        try:
+            result = EXPERIMENTS_MAP[experiment.lower().replace(' ', '')]
```

Comment:
This seems like in most of the cases does just `experiment.lower().replace(' ', '').upper()` (at least for most of the experiments in the map)